### PR TITLE
Updated document factory implementation to use regular expression to …

### PR DESCRIPTION
**Pull request by sticklerdev**
…validate spec version. Added unit tests for a few other conditions.

### CODE REVIEW

File: src/factories/document.factory.ts [hunk 1]
```diff
@@ -102,4 +98,38 @@ export class OasDocumentFactory {
...
     }
 
-}
\ No newline at end of file
+    /**
+     * @param ver 
+     * @return {string | undefined}
+     */
+    private parseVersion(ver: any) {
+        let version;
```
Comment (EricWittmann) [inline src/factories/document.factory.ts]: Can you add a type to the `version` variable?

File: src/factories/document.factory.ts [hunk 2]
```diff
@@ -26,30 +26,31 @@ import {Oas30Document} from "../models/3.0/document.model";
...
     /**
      * Creates a new, empty instance of an OAS document.
      * @param oasVersion
      * @return {OasDocument}
      */
     public createEmpty(oasVersion: string): OasDocument {
-        if (typeof oasVersion === "number") {
-            oasVersion = "" + oasVersion;
-        }
+        const ver = oasVersion;
```
Comment (EricWittmann) [inline src/factories/document.factory.ts]: Add a type to `ver`... probably should be `string` based on the input param type, but could also perhaps be `string|number`.  Dealer's choice. :)  I just like having types for things when possible (not always possible).

**Comment by sticklerdev** [comment]
I have made the two changes you requested. Sorry - have not done this before so have a question? Should I simply push my changes to the 'improved-spec-version-checks' branch on my fork or will I have to create another pull request? Thanks.

**Comment by EricWittmann** [comment]
Just push to the same branch and GH should figure everything out. :)

**Comment by EricWittmann** [comment]
Nice, thanks!

**Pull request merged**